File: Xamarin101/TaskyCalabash/features/utilities/common_values.rb
module CommonValues
  very_long_wait = 60
  long_wait = 30
  moderate_wait = 15
  short_wait = 10
  very_short_wait = 5

  long_interval = 1
  moderate_interval = 0.5
  short_interval = 0.3

  def waiting_for_element_timed_out_message(element)
    "Timed out waiting for #{element}"
  end

  def waiting_for_element_to_disappear_timed_out_message(element)
    "Timed out waiting for #{element} to disappear"
  end
end

File: Xamarin101/Tasky/TaskyAndroid.UITest/Utilities/CommonValues.cs
using System;

namespace TaskyAndroid.UITest
{
	public static class CommonValues
	{
		public readonly static TimeSpan VeryLongWait = TimeSpan.FromSeconds (60);
		public readonly static TimeSpan LongWait = TimeSpan.FromSeconds (30);
		public readonly static TimeSpan ModerateWait = TimeSpan.FromSeconds (15);
		public readonly static TimeSpan ShortWait = TimeSpan.FromSeconds (10);
		public readonly static TimeSpan VeryShortWait = TimeSpan.FromSeconds (5);

		public readonly static TimeSpan VeryLongInterval = TimeSpan.FromMilliseconds (1000);
		public readonly static TimeSpan LongInterval = TimeSpan.FromMilliseconds (500);
		public readonly static TimeSpan ModerateInterval = TimeSpan.FromMilliseconds (300);
		public readonly static TimeSpan ShortInterval = TimeSpan.FromMilliseconds (200);
	}

	public static class Messages
	{
		public static string WaitingForElementTimeoutMessage (string element)
		{
			return "Timed out waiting for " + element;
		}
	}
}

